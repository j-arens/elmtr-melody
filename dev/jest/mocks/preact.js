module.exports = {
    unmountComponent: jest.fn(),
};
