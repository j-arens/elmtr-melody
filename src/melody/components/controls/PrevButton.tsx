import BaseButton from '@components/BaseButton/';
import Icon from '@components/Icon/';
import { prevTrack } from '@redux/actions';
import { Action, State } from '@redux/type';
import { NO_OP } from '@utils/index';
import { h } from 'preact';
import { connect } from 'preact-redux';
const s = require('./styles.scss');

const mapStateToProps = (state: State) => ({
    totalTracks: state.tracks.length,
});

const mapDispatchToProps = dispatch => ({
    prevTrack: () => dispatch(prevTrack()),
});

interface Props {
    totalTracks: number;
    prevTrack: () => Action;
}

const PrevButton = ({ totalTracks, prevTrack }: Props) => {
    const defaultClass = `${s.playbackCtrl} melody-playbackCtrl`;
    const disabledClass = `${defaultClass} ${s['playbackCtrl--disabled']}`;
    return (
        <BaseButton
            key="prev"
            className={totalTracks > 1 ? defaultClass : disabledClass}
            onClick={totalTracks > 1 ? prevTrack : NO_OP}
        >
            <Icon className={s.playbackCtrl__icon} name="prev" />
        </BaseButton>
    );
};

export default connect(mapStateToProps, mapDispatchToProps)(PrevButton);
