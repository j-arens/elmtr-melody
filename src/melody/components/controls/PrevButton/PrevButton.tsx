import BaseButton from '@components/BaseButton/';
import Icon from '@components/Icon/';
import { WithOptionalClassName } from '@melody/components/type';
import { Action } from '@redux/type';
import { h } from 'preact';
const s = require('../styles.scss');

export interface DispatchProps {
    prevTrack: () => Action;
}

type Props = DispatchProps & WithOptionalClassName;

export default ({ prevTrack, className = '' }: Props) => (
    <BaseButton
        key="prev"
        className={`${className} ${s.playbackCtrl} melody-playbackCtrl`}
        onClick={prevTrack}
    >
        <Icon className={s.playbackCtrl__icon} name="prev" />
    </BaseButton>
);
