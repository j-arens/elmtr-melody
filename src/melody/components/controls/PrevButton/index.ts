import { WithOptionalClassName } from '@components/type';
import { prevTrack } from '@redux/modules/audio/actions';
import { State } from '@redux/type';
import { connect } from 'preact-redux';
import { bindActionCreators } from 'redux';
import PrevButton, { DispatchProps } from './PrevButton';

const mapDispatch = dispatch => bindActionCreators({
    prevTrack,
}, dispatch);

export default connect<
    {},
    DispatchProps,
    WithOptionalClassName
>(null, mapDispatch)(PrevButton);
