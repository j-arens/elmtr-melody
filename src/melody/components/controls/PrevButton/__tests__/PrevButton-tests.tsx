import { h } from 'preact';
import { render } from 'preact-render-to-string';
import PrevButton from '../PrevButton';

describe('PrevButton', () => {
    it('renders', () => {
        const component = render(
            <PrevButton
                totalTracks={100}
                className="sapphire-blue"
            />,
        );
        expect(component).toMatchSnapshot();
    });

    it('renders with a disabled class if totalTracks <= 1', () => {
        const component = render(
            <PrevButton
                totalTracks={1}
                className="sapphire-blue"
            />,
        );
        expect(component).toMatchSnapshot();
    });
});
