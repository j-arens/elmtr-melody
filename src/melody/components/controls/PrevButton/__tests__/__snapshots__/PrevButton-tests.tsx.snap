// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrevButton renders 1`] = `
<button data-cy="ctrl prev" class="btn sapphire-blue playbackCtrl melody-playbackCtrl"
role="button">
  <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
    <path d="M11 18V6l-8.5 6 8.5 6zm.5-6l8.5 6V6l-8.5 6z"></path>
  </svg>
</button>
`;

exports[`PrevButton renders with a disabled class if totalTracks <= 1 1`] = `
<button data-cy="ctrl prev" class="btn sapphire-blue playbackCtrl melody-playbackCtrl"
role="button">
  <svg class="playbackCtrl__icon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
    <path d="M11 18V6l-8.5 6 8.5 6zm.5-6l8.5 6V6l-8.5 6z"></path>
  </svg>
</button>
`;
